source/event_stream.c

```diff
@@ -1013,7 +1013,8 @@ static int s_read_header_value(
     size_t length_read = current_pos - decoder->current_header_value_offset;
     struct aws_event_stream_header_value_pair *current_header = &decoder->current_header;
 
-    if (!length_read) {
+    if (!length_read && (current_header->header_value_type == AWS_EVENT_STREAM_HEADER_BYTE_BUF ||
+                         current_header->header_value_type == AWS_EVENT_STREAM_HEADER_STRING)) {
         /* save an allocation, this can only happen if the data we were handed is larger than the length of the header
```

Comment:
how did this ever work before?

the old code would end up reading an arbitrary memory address as the value of an integer.... how did that ever pass any test?